### PR TITLE
Replace section of Explore about RPiOS images with Yocto

### PR DESCRIPTION
**Pull request by manuelafm**
In recent meetings (e.g. WebKit Codecamp Oct 2022) it was decided to drop them, as they are unfavourable compared to the Yocto images, specially in terms of showing performance.

Besides, some parts of the text are clearly out of date, like mentioning $45 as the price of a RPi 3, which is currently not even possible to obtain due to supply shortages.

----

Site preview: https://igalia.github.io/wpewebkit.org/mafm-update-explore/

### CODE REVIEW

**Review by psaavedra — APPROVED** [review]

File: about/yocto-wpe.md
Comment (mariospr) [inline about/yocto-wpe.md:3]: I think this title should be shortened a bit, otherwise it does not look great on the page, splitting "Pi" to a new line:

image

How about just "Yocto images for the Raspberry Pi"?

Comment (manuelafm) [reply about/yocto-wpe.md:3]: OK to shortening but maybe we should keep WPE for more relevant web searches?  (I'm not a SEO or anything :stuck_out_tongue_closed_eyes: , but intuitively, should be better)

Like "WPE Yocto images for Raspberry Pi".

Comment (mariospr) [reply about/yocto-wpe.md:3]: Oh! I see we have another section titled "Try WPE with Balena", so my suggestion would be inconsistent... maybe we can omit the "Yocto" word in the title and just leave "Try WPE with images for Raspberry Pi"? (after all, whether the images are created with yocto or something else is an implementation detail).

File: about/yocto-wpe.md
Comment (mariospr) [inline about/yocto-wpe.md:33]: This is a bit confusing... I understand how the sustitutions work, but couldn't we just provide the final URLs here so that users only have to click on them (or copy & paste the `wget` command) instead of having to manually do the substitutions first?

If we only have two images that would be only `wget` lines, right?

File: about/yocto-wpe.md
Comment (mariospr) [inline about/yocto-wpe.md:57]: I think this won't work unless you either add an `export` to the beginning of the first two lines, or put the three lines in one single command line, separated by spaces.

If, in the other hand, what you mean is that the user needs to do the substitutions manually, then I think it would be better to simply provide the two lines for 32bit and 64bit images, as I mentioned above.

File: about/yocto-wpe.md
Comment (mariospr) [inline about/yocto-wpe.md:92]: There are only four environment variables in https://github.com/Igalia/cog/blob/master/docs/platform-wl.md so I think we could somehow summarize them here right away and then simply refer to the URL as the place to get more information, without using a single bullet item for it (which looks a bit awkward IMHO).

How about something like this:

```
There are also some environment variables that can be used to change the behaviour of Cog's window (e.g.
fullscreen, maximized, specific dimensions), as well as some key bindings that can be used to perform certain
actions with Cog or change its behaviour (e.g. toggle fullscreen, reload, zooming, exit...), when running on
Wayland. You can check more details about this in [https://github.com/Igalia/cog/blob/master/docs/platform-wl.md](https://github.com/Igalia/cog/blob/master/docs/platform-wl.md).
```

File: about/yocto-wpe.md
Comment (mariospr) [inline about/yocto-wpe.md:112]: Would it make sense to duplicate the resources and put them under `../assets/yocto-images/`, or even to create new ones off the actual Yocto images instead of re-using the Balena ones?

**Review by mariospr — CHANGES_REQUESTED** [review]
@manuelafm This is looking great, thanks for working on this!

I have a couple of small comments to consider before merging but nothing out of the ordinary I think. Also, I think it would be worth running this by @bkardell or @meyerweb if possible too before merging, for an extra sanity check on possible language-related bits that could perhaps be improved.

File: about/yocto-wpe.md
Comment (manuelafm) [reply about/yocto-wpe.md:57]: > Also, I think it would be worth running this by @bkardell or @meyerweb if possible too before merging, for an extra sanity check on possible language-related bits that could perhaps be improved.

@bkardell is added as reviewer (by default I think) but I'll make sure to ping him.

File: about/yocto-wpe.md
Comment (manuelafm) [reply about/yocto-wpe.md:33]: Yes, initially we had more `wget`s because of the Wandboard images, but since Pablo suggested to remove them this part is probably more noise than signal.

File: about/yocto-wpe.md
Comment (manuelafm) [reply about/yocto-wpe.md:57]: This works fine without `export`, `exports` are only needed for children processes or something, but not when executing `wget` immediately:
```
$ MACHINE=raspberrypi3-mesa
$ SDIMG="core-image-weston-browsers-${MACHINE}.wic"
$ wget -nv https://wk-contrib.igalia.com/yocto/meta-webkit/browsers/stable/images/${MACHINE}/${SDIMG}.{bz2,bmap}
2023-01-10 11:06:34 URL:https://wk-contrib.igalia.com/yocto/meta-webkit/browsers/stable/images/raspberrypi3-mesa/core-image-weston-browsers-raspberrypi3-mesa.wic.bz2 [374408403/374408403] -> "core-image-weston-browsers-raspberrypi3-mesa.wic.bz2" [1]
2023-01-10 11:06:34 URL:https://wk-contrib.igalia.com/yocto/meta-webkit/browsers/stable/images/raspberrypi3-mesa/core-image-weston-browsers-raspberrypi3-mesa.wic.bmap [3506/3506] -> "core-image-weston-browsers-raspberrypi3-mesa.wic.bmap" [1]
FINISHED --2023-01-10 11:06:34--
Total wall clock time: 27s
Downloaded: 2 files, 357M in 27s (13.2 MB/s)
```

Anyway, as explained above: this was maybe more useful when having more machines like with the Wandboard that I had initially, and maybe in the future becomes relevant again, but at the moment is probably better to remove this part.

File: about/yocto-wpe.md
Comment (manuelafm) [reply about/yocto-wpe.md:92]: ack

File: about/yocto-wpe.md
Comment (manuelafm) [reply about/yocto-wpe.md:112]: I think that recreating this would be busywork for nothing.  These are not "accurate representations of execution with Yocto images", like in a video, when for example you can see the framerate or if there was some effect/feature that could be seen with Yocto that couldn't be seen with the Balena one.

Basically the images were already there, and I thought that the page would be just more pretty with the pictures, I think that the page is better this way than only with text.

But I think that replicating the images with Yocto is not worth the effort or that we should stop for doing this right now, IMHO... maybe we can revisit this in the future if/when adding new/more demos or something.

File: about/yocto-wpe.md
Comment (mariospr) [reply about/yocto-wpe.md]: > This works fine without `export`, `exports` are only needed for children processes or something, but not when executing `wget` immediately:

Oh! Sorry for the misleading comment... I was completely convinced an export would be needed, TIL...

Comment (mariospr) [reply about/yocto-wpe.md]: > So I think that in this part, specially with future considerations in mind, we should preserve the variables, WDYT?

I guess it works, especially now I realized `export` is not needed (:facepalm: ) but I'm still not very excited about having this, I'd rather have a link to the actual files as we only need two at the moment.

And in the future, if we have more images we can perhaps think of something slightly better and more scalable, like a bunch of combo boxes + some JavaScript that allows you to generate the download link and/or the snippet you need to use to get the images on the fly (e.g. something like what you can see in nvidia's website at https://www.nvidia.com/download/index.aspx, but simpler).

File: about/yocto-wpe.md
Comment (mariospr) [reply about/yocto-wpe.md:111]: I understand what yo mean but still I think we should aim for having something better there than those animations anyway, they feel a bit too "hackish" to be in the official site of WPE to be honest.

For instance, I think we should aim for having proper screen recordings taken in the devices, just showing the desktop and Cog running inside of it. I don't think there's any real value on having a low quality video recorded with an external camera (a phone even?) showing the actual screen and the rpi running the WPE image beyond assuring the viewer that the image is running on the actual device... but even in that case the current gif animations are not showing the actual thing since they are showcasing the balena images anyway.

Anyway, I won't block the PR for this, but I think we should produce better images/videos (i.e. better quality, better framerate) if we are going to use them here, but that can be a follow up PR.

File: about/yocto-wpe.md
Comment (mariospr) [inline about/yocto-wpe.md:20]: I think this should be written in a different way, maybe split in two different lines or structured differently, otherwise it's hard to read on the website:

image

File: about/yocto-wpe.md
Comment (manuelafm) [reply about/yocto-wpe.md:111]: > I don't think there's any real value on having a low quality video recorded with an external camera (a phone even?) showing the actual screen and the rpi running the WPE image beyond assuring the viewer that the image is running on the actual device...

But if we only capture the screen then we cannot see @psaavedra 's hand playing with BouncyBalls... :stuck_out_tongue_winking_eye: 

More seriously... I take the point of being a bit hackish and so on, with the camera moving etc, but OTOH the ones with the hand interacting show things than a simple screen capture wouldn't (also with the set-up of the device, screen, etc).  One can even appreciate the smoothness of scrolling with the old website, for example, and the immediacy of the stream starting playing when the finger presses.

It would be worth considering what we really want to show, to see if we use simple images (or nothing at all), or small videos, or what.

> [...] but even in that case the current gif animations are not showing the actual thing since they are showcasing the balena images anyway.

(FWIW: IIRC the binaries used in Balena are actually created in the same way, so it's basically the same thing)

Comment (mariospr) [reply about/yocto-wpe.md:90]: Like I said in the morning in a call, I think these demonstrations can be still improved, which is important if we think this website is product-oriented (vs developers-oriented), to properly showcase WPE.

Like I said, I don't see that the value of seeing the actual stuff (screen, board, fingers...) is more important than providing higher quality recordings of Cog performing on a RaspberryPi board, which for me it's more prioritary. Also, it's also arguable how smoothness can be appreciated on a low-FPS gif animation... from my POV, we should aim for having higher quality mini-demos here, and if we want to show the actual devices then let's do it properly with a decent recording setup: e.g. a good camera on a tripod, good framing of the scene, good lightning...

In any case, this is not a blocker, I'm Ok with merging this as-is now, but I think we should work on better mini-demos in a follow-up PR, that is on par with the kind of product-oriented quality we want  on wpewebkit.org.

File: about/yocto-wpe.md
Comment (mariospr) [inline about/yocto-wpe.md:38]: Can we add a note like _" (replace `mmcblk_XYZ` with your actual target device)"_, right before the colon?

File: about/yocto-wpe.md
Comment (mariospr) [inline about/yocto-wpe.md:40]: Could we split this line in two, and be more clear about the contents? How about this:
```
bzip2 -dc core-image-weston-browsers-raspberrypi3-mesa_202205_4.0_kirkstone_r1_20221228073347.wic.bz2
sudo dd if=core-image-weston-browsers-raspberrypi3-mesa_202205_4.0_kirkstone_r1_20221228073347.wic of="/dev/mmcblk_XYZ" bs=4k status=progress
```

**Review by mariospr — APPROVED** [review]
LGTM, with just a couple of minor changes I think would be good to make around the instructions for flashing

File: about/yocto-wpe.md
Comment (manuelafm) [reply about/yocto-wpe.md:40]: The idea of doing it with the pipe was to avoid littering the place with uncompressed copies around and be slightly faster, but OK.

**Comment by manuelafm** [comment]
> LGTM, with just a couple of minor changes I think would be good to make around the instructions for flashing

I don't seem to be able to reply directly to the bit about the demos, not sure why, so replying here:

> In any case, this is not a blocker, I'm Ok with merging this as-is now, but I think we should work on better mini-demos in a follow-up PR, that is on par with the kind of product-oriented quality we want on wpewebkit.org

We have the real pretty and well produced demo videos in:
https://wpewebkit.org/about/#wpe-in-action

This section (copied from the Balena page) was meant to have links to demos for immediate testing, and some pictures so the longish page of dry-text was not so dry.  But if we only want the "pro" demos, having the links here doesn't seem very necessary either, maybe we can drop this subsection completely, both from the Yocto page and the Balena one?

**Comment by mariospr** [comment]
> This section (copied from the Balena page) was meant to have links to demos for immediate testing, and some pictures so the longish page of dry-text was not so dry. But if we only want the "pro" demos, having the links here doesn't seem very necessary either, maybe we can drop this subsection completely, both from the Yocto page and the Balena one?

I think that would be a good idea, yes. After all this page is just about instructions for testing, no need to duplicate efforts.

**Comment by mariospr** [comment]
All good, merged!

**Pull request merged**
